Added 14 Logos and Optimised a few files

dbkonxepts opened the pull request:
**Added 14 Logos**
```
1. Business Day
2. Circa
3. Disrupt Africa
4. Federal Inland Revenue Service
5. Investment One
6. Nigeria Football Federation
7. National Identity Management Commission
8. Oando
9. ThankUCash
10. The Tony Elumelu Foundation
11. Total
12. WalletNG
13. Wilson's
14. Workstation
```

**Optimised search.js**  
```
So that it is more functional and can search categories too, not just company name
```
**Optimised imageLoad.js** 
```
So that it loads singular categories. This helps those whose company category fall in a group that they don't necessarily belong.
```
**Edited contributing.md** 
```
So it includes a better instruction based on the new changes and also added new categories
```

Odusanya commented:
Hey Man @dbkonxepts, thanks for being consistent with this 💪🏽💪🏽💪🏽.

A few things

1.) **Search Optimization**
Really appreciate that you took the extra time to think about optimizing the way search works, best guy! ✨. The only issue that comes to mind is that I can't find an immediate use case for why users would want the search field to filter both categories and company names at the same time. 

On the contrary, I think it introduces more complexity to the user's flow, especially because this feature becomes non-applicable when the categories select is no longer set to `All Categories`. 
For example, Say a user has already selected Banking from the dropdown subconsciously and types another category name out of ignorance, they would get **0** results. There's no way to explicitly communicate when we're showing results for categories vs when we're showing results for Company Names.


Also, tiny thing: If we're eventually going to end up with having the input filter through both categories & name, I would suggest we update the placeholder text to `Search for a company or category` - It currently says `Search for a company`.


2.) ** ImageLoad **
I'll leave comments on the file.

dbkonxepts commented:
@Odusanya Thanks for the feedback!
Alright I think I see your point.  Since I have streamlined the category filter to singular list there's no need to be able to search by category too. _I will take out the ability to filter category in search box_

Odusanya commented on js/imageLoad.js:
```diff
@@ -32,7 +32,15 @@ function createLogos(logoArray) {
+    const rawcategories = Object.values(map).sort();
+    const categories = []; 
+    rawcategories.forEach(category => {
+        category.replace(" /", "/").replace("/ ","/") //Avoided replace(" / ") so all cases are checked and replaced
```
My immediate first impression was, "Woow! This dropdown is long oh! Imagine the experience on a mobile phone" Then I realized it actually something that's inevitable in the end.

WRT this implementation - this is just out of curiosity. Any particular reason why you didn't update the category key to accept an array, instead of separating with a `/`?

We would have stuff like this
```
{
	"title": "ALAT by Wema",
	"filename": "alat_by_wema",
	"category": ["Banking" , "Financial Services"]
}
```

dbkonxepts replied:
Hmmm. This actually makes sense.  
_It's doable_

Odusanya replied:
Thanks a lot bro! 💪🏽💪🏽💪🏽

dbkonxepts replied:
Quick One: How do suggest, the category be displayed on the landing page?

Odusanya replied:
@dbkonxepts Just to be sure, I get your question, you're asking how I think the categories should be listed in this dropdown down right?
![image](https://user-images.githubusercontent.com/22182946/53807605-48091880-3f50-11e9-8e90-e2d9534895f3.png)

dbkonxepts replied:
Not that. The individual logo category. If we go with array approach.

dbkonxepts commented:
Mistakenly closed the PR  _-^^-_

dbkonxepts commented:
1. Cleaned up the **imageLoad.js** _for better code management and optimisation._
2. Removed the ability to search categories in **search.js** _since it has been made in a singular form on the drop down._
3. Moved script files to bottom in **index.html** _for better and faster loading_
4. Changed the category key of each jsonObject in **logo.json** _so its now an array._
4. Edited the **contributing.md** to reflect the changes **logo.json**

fathermerry commented on index.html:
```diff
@@ -45,40 +46,37 @@
-                        <!-- OTHER CATEGORIES AUTOGENERATED HERE 
-                            Format: Same as the option above.
-                        -->
+                        <!-- OTHER CATEGORIES AUTOGENERATED HERE Format: Same as the option above.-->
```
```suggestion
                        <!-- CATEGORIES ARE AUTOMATICALLY POPULATED HERE -->
```
